src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -267,8 +267,9 @@ async ValueTask<int> ReadAheadWithZeroByteReadAsync()
 
                     return read;
                 }
-                catch (Exception error) when (TransitionToCompletedAndTryOwnCompletion())
+                catch (Exception error)
```

Comment:
`TransitionToCompletedAndTryOwnCompletion` controls who is responsible for observing the exception.
If a thread has already flipped `_readAheadTaskStatus` to `CompletionReserved`, it's also responsible for observing any exceptions thrown by this method.
The exact exception is meaningful as the request may fail with it as the inner exception.

If it has not (task status is still `Started`), the method itself will observe and suppress the exception. In that case we don't necessarily care that/which exception occured, just that the read completed before a request attempted to use the connection.